packages/ember-glimmer/tests/integration/components/fragment-components-test.js

```diff
@@ -0,0 +1,156 @@
+
+    expectAssertion(() => {
+      this.render(`{{#foo-bar}}{{/foo-bar}}`);
+    }, /You can not define a function that handles DOM events in the .* tagless component since it doesn't have any DOM element./);
+  }
+
+  ['@htmlbars throws an error if a custom defined event function is defined in a tagless component']() {
+    let instance;
+    let template = `hit dem folks`;
+    let FooBarComponent = Component.extend({
+      tagName: '',
+      init() {
+        this._super();
+        instance = this;
+      },
+      folks() { }
+    });
+
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template });
+
+    expectAssertion(() => {
+      this.render(`{{#foo-bar}}{{/foo-bar}}`);
+    }, /You can not define a function that handles DOM events in the .* tagless component since it doesn't have any DOM element./);
+  }
+
+  ['@htmlbars throws an error if `tagName` is an empty string and `classNameBindings` are specified']() {
+    let instance;
+    let template = `hit dem folks`;
+    let FooBarComponent = Component.extend({
+      tagName: '',
+      init() {
+        this._super();
+        instance = this;
+      },
+      foo: true,
+      classNameBindings: ['foo:is-foo:is-bar']
+    });
+
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template });
+
+    expectAssertion(() => {
+      this.render(`{{#foo-bar}}{{/foo-bar}}`);
+    }, /You cannot use `classNameBindings` on a tag-less component/);
+  }
+
+  ['@htmlbars throws an error if when $() is accessed on component where `tagName` is an empty string']() {
+    let template = `hit dem folks`;
+    let FooBarComponent = Component.extend({
+      tagName: '',
+      init() {
+        this._super();
+        this.$();
+      }
+    });
+
+    this.registerComponent('foo-bar', { ComponentClass: FooBarComponent, template });
+
+    expectAssertion(() => {
+      this.render(`{{#foo-bar}}{{/foo-bar}}`);
+    }, /You cannot access this.\$\(\) on a component with `tagName: \'\'` specified/);
+  }
+
+  ['@htmlbars renders a contained view with omitted start tag and tagless parent view context']() {
+    this.registerComponent('root-table', {
+      ComponentClass: Component.extend({
+        tagName: 'table'
+      }),
+      template: '{{frag-ment}}'
+    });
+
+    this.registerComponent('frag-ment', {
+      ComponentClass: Component.extend({
+        tagName: ''
+      }),
+      template: '{{my-row}}'
+    });
+
+    this.registerComponent('my-row', {
+      ComponentClass: Component.extend({
+        tagName: 'tr'
+      }),
+      template: 'dab'
+    });
+
+    this.render(`{{root-table}}`);
+
+    this.assertElement(this.firstChild, { tagName: 'table' });
+    this.assertElement(this.firstChild.firstElementChild, { tagName: 'tr' });
+
+    this.runTask(() => this.rerender());
+
+    this.assertElement(this.firstChild, { tagName: 'table' });
+    this.assertElement(this.firstChild.firstElementChild, { tagName: 'tr' });
```

Comment:
Can we use a different tag combo? this html wouldn't SSR correctly and rehydrate because of the missing tbody, I feel weird about codifying it in a test. I know it was ported from an existing test, but I think that was also originally testing if the old morph tags survived the auto tbody insertion.  This no longer seems relevant.